### ISSUE TITLE
weblogic_deserialize_asyncresponseservice check is not accurate

### ISSUE THREAD

**Issue by red-1337er** [rapid7/metasploit-framework#12247]
The check part of the weblogic_deserialize_asyncresponseservice module (exploit/multi/misc/weblogic_deserialize_asyncresponseservice) does not accurately identify a vulnerable system.

Currently, the "check" is defined as:

![image](https://user-images.githubusercontent.com/33378226/63894638-f9d64e80-c9a1-11e9-8df2-8bd6d7b0dae6.png)

It's essentially sending a POST to the specified URI (Target IP + URI Path), and matching up the server response with the specified IF statements to determine whether the target system is vulnerable.

If you look at the highlighted portion, it's essentially stating that IF the server response with a status code of 500 AND the response contains the string "<faultcode>env:Client</faultcode>", then the target system IS vulnerable.

I tested this against an unpatched instance of WebLogic 12.1.3.0.0. The "check" came back as vulnerable, and I was able to gain shell using the exploit.

I tested this again after applying the patch for the vulnerability. The "check" still came back as vulnerable, but I was not able to gain shell with the exploit after trying numerous different payloads.

In short, it looks as though a patched instance of WebLogic will continue to be classified as vulnerable via the "check" as the patch does not alter how the server responds to the POST request.

**Comment by bwatters-r7** [comment]
Can you tell us what the difference in the response was between the patched and unpatched systems?  That would simplify the correction of this bug.

**Comment by red-1337er** [comment]
Running check on unpatched system:
![image](https://user-images.githubusercontent.com/54647493/66159526-066f4780-e5dd-11e9-8fff-df8241d39362.png)

Running check on patched system:
![image](https://user-images.githubusercontent.com/54647493/66159600-31f23200-e5dd-11e9-9f75-891d3383d08f.png)

**Comment by dwelch-r7** [comment]
@red-1337er it hard to check by eye but it looks like the check function has printed out the same thing there for both patched and unpatched
Are you able to get the full response from the patched vs the unpatched system an we can see if there is a difference there

**Comment by red-1337er** [comment]
@dwelch-r7 this is all that the check provides

**Comment by space-r7** [comment]
Hey @red-1337er, is there any chance you still have access to this system? If so, is there any chance you could run the `check` method against the patched system with `HttpTrace => true`? Thanks so much!

**Comment by ccondon-r7** [comment]
Without the patch that's provided for paid customers, the team could only test newer releases. In fixed versions of WebLogic, the particular request used in the check always results in a 404 response code, which is consistent with the module's check method. Closing this for now; if OP or someone with access to a patched system can get back to @space-r7 on her request above, we can reopen.
